### PR TITLE
Liav/cert 5572 second challenge

### PR DESCRIPTION
**Pull request by liav-certora**

### CODE REVIEW

File: contracts/naive-receiver/Challenge2.spec [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    address borrower;
+    uint256 amount;
+    address otherUser;
+
+    uint256 otherUserBalanceBefore = nativeBalances[otherUser];
+    flashLoan(borrower, amount);
+
+    assert nativeBalances[otherUser] != otherUserBalanceBefore => otherUser == e.msg.sender;
+}
+*/
```
Comment (nd-certora) [inline contracts/naive-receiver/Challenge2.spec]: remove this rule

File: contracts/naive-receiver/Challenge2.spec [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+rule noChangeToOtherUser(method f){// filtered { f -> f.contract == currentContract } {
```
Comment (nd-certora) [inline contracts/naive-receiver/Challenge2.spec]: remove comment

File: contracts/naive-receiver/Challenge2Fixed.conf [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    "files": [
+        "contracts/naive-receiver/FlashLoanReceiver.sol",
+        "contracts/naive-receiver/NaiveReceiverLenderPoolFixed.sol:NaiveReceiverLenderPool",
+    ],
+    "verify": "NaiveReceiverLenderPool:contracts/naive-receiver/Challenge2.spec",
+    "link": [
+        "FlashLoanReceiver:pool=NaiveReceiverLenderPool"
+    ],
+    "optimistic_loop": true,
+    "solc": "solc8.20"
```
Comment (nd-certora) [inline contracts/naive-receiver/Challenge2Fixed.conf:11]: we should avoid solc, must users use solc-select and that has a different naming scheme

File: contracts/naive-receiver/Readme.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Challenge 2 - Naive Receiver
+There’s a pool with 1000 ETH in balance, offering flash loans. It has a fixed fee of 1 ETH.
+A user has deployed a contract with 10 ETH in balance. It’s capable of interacting with the pool and receiving flash loans of ETH.
+Take all ETH out of the user’s contract. If possible, in a single transaction.
+
+### Finding The Bug
+Using the prover we wrote a parametric rule `noChangeToOtherUser` that checks that no interaction with the pool can result in
```
Comment (nd-certora) [inline contracts/naive-receiver/Readme.md]:
```suggestion
A rather common parametric rule `noChangeToOtherUser` checks that no interaction with the pool can result in
```

**Review by nd-certora — APPROVED** [review]
notice comments

File: contracts/naive-receiver/Challenge2.spec (hunk 1 above)
Comment (liav-certora) [reply contracts/naive-receiver/Challenge2.spec]: removed

File: contracts/naive-receiver/Challenge2.spec (hunk 2 above)
Comment (liav-certora) [reply contracts/naive-receiver/Challenge2.spec]: removed

File: contracts/naive-receiver/Challenge2Fixed.conf (hunk 3 above)
Comment (liav-certora) [reply contracts/naive-receiver/Challenge2Fixed.conf:11]: removed

**Pull request merged**